packages/core/src/odata-common/selectable/collection-field.ts

```diff
@@ -1,24 +1,18 @@
 import { Entity } from '../entity';
 import { EdmTypeShared } from '../edm-types';
-import { SelectableEdmTypeField } from './edm-type-field';
+import { SelectableField } from './selectable';
 import { Field } from './field';
-import { ComplexTypeNamespace } from './complex-type-namespace';
-import { getEntityConstructor } from './complex-type-field';
 import { ConstructorOrField } from './constructor-or-field';
+import { getEntityConstructor } from './complex-type-field';
+import { ComplexTypeNamespace } from './complex-type-namespace';
 
-/**
- *
- * Represents a static field of an entity or complex type.
- *
- * @typeparam EntityT - Type of the entity the field belongs to
- * @typeparam FieldT - Type of the entries of the collection in the field
- */
 export class CollectionField<
     EntityT extends Entity,
-    FieldT extends EdmTypeShared<'any'> | Record<string, any> = any
+    CollectionFieldT extends EdmTypeShared<'any'> | Record<string, any> = any,
```

Comment:
Also this type feels like a property encoding like `isNullable` so perhaps this should also go to a generic property?

Reply:
Done.